status/reporters.go

```diff
@@ -445,133 +354,73 @@ func (r *GitHubReporter) createCommitStatusAdapterForSnapshot(snapshot *applicat
+		pipelineRunName := integrationTestStatusDetail.TestPipelineRunName
+		pipelineRun := &tektonv1beta1.PipelineRun{}
+		err := r.k8sClient.Get(ctx, types.NamespacedName{
+			Namespace: snapshot.Namespace,
+			Name:      pipelineRunName,
+		}, pipelineRun)
 		if err != nil {
-			return err
+			return fmt.Errorf("error while getting the pipelineRun %s: %w", pipelineRunName, err)
 		}
 
-		token, err := r.client.CreateAppInstallationToken(ctx, creds.AppID, creds.InstallationID, creds.PrivateKey)
+		taskRuns, err := helpers.GetAllChildTaskRunsForPipelineRun(r.k8sClient, ctx, pipelineRun)
 		if err != nil {
-			return err
+			return fmt.Errorf("error while getting all child taskRuns from pipelineRun %s: %w", pipelineRunName, err)
 		}
-
-		r.client.SetOAuthToken(ctx, token)
-
-		checkRun, err := r.createCheckRunAdapter(k8sClient, ctx, pipelineRun)
+		comment, err = FormatCommentForFinishedPipelineRun(title, taskRuns)
 		if err != nil {
-			return err
-		}
-
-		checkRunID, err := r.client.GetCheckRunID(ctx, checkRun.Owner, checkRun.Repository, checkRun.SHA, checkRun.ExternalID, creds.AppID)
-		if err != nil {
-			return err
-		}
-
-		if checkRunID == nil {
-			_, err = r.client.CreateCheckRun(ctx, checkRun)
-		} else {
-			err = r.client.UpdateCheckRun(ctx, *checkRunID, checkRun)
-		}
-
-		if err != nil {
-			return err
+			return fmt.Errorf("error while formating all child taskRuns from pipelineRun %s: %w", pipelineRun.Name, err)
 		}
 	} else {
-		token, err := r.getToken(ctx, pipelineRun, pipelineRun.Namespace)
+		comment, err = FormatCommentForDetail(title, integrationTestStatusDetail.Details)
 		if err != nil {
 			return err
 		}
+	}
 
-		r.client.SetOAuthToken(ctx, token)
-
-		err = r.createCommitStatus(k8sClient, ctx, pipelineRun)
+	allComments, err := r.client.GetAllCommentsForPR(ctx, owner, repo, issueNumber)
+	if err != nil {
+		return fmt.Errorf("error while getting all comments for pull-request %s: %w", issueNumberStr, err)
+	}
+	existingCommentId := r.client.GetExistingCommentID(allComments, snapshot.Name, integrationTestStatusDetail.ScenarioName)
+	if existingCommentId == nil {
+		_, err = r.client.CreateComment(ctx, owner, repo, issueNumber, comment)
 		if err != nil {
-			return err
+			return fmt.Errorf("error while creating comment for pull-request %s: %w", issueNumberStr, err)
 		}
-
-		err = r.createComment(k8sClient, ctx, pipelineRun)
+	} else {
+		_, err = r.client.EditComment(ctx, owner, repo, *existingCommentId, comment)
 		if err != nil {
-			return err
+			return fmt.Errorf("error while updating comment for pull-request %s: %w", issueNumberStr, err)
 		}
 	}
 
 	return nil
 }
 
-// ReportStatusForSnapshot creates CheckRun when using GitHub App integration,
-// creates a a commit status when using GitHub webhook integration
+// ReportStatusForSnapshot creates CheckRun when using GitHub App integration.
+// When using GitHub webhook integration a commit status and a comment.
```

Comment:
nit: `When using GitHub webhook integration a commit status and a comment.` -> `When using GitHub webhook integration it creates a commit status and a comment.`

Reply:
Updated. Thanks.